### PR TITLE
add shellcheck static checks and make all scripts pass them

### PR DESCRIPTION
**Pull request by mikedlr**
This adds shellcheck based static tests and modifies the scripts so they pass (with a couple of exceptions where checks have been disabled). Noticeable number of changes in the scripts from spamgourmet-clone since that's our biggest set of shell scripts.

### CODE REVIEW

File: spamgourmet-clone/scripts/config-certs-provisioning.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ echo '### set up automated provisioning for exim4 and lighttpd'
 echo '##########################################################################'
 source sg-server-config.sh
 
-cd $SCRIPT_BASE_DIR
+cd "$SCRIPT_BASE_DIR" || exit 72
```
Comment (fredericmora) [inline spamgourmet-clone/scripts/config-certs-provisioning.sh:8]: Why the exit code 72? Is it documented somewhere?

File: spamgourmet-clone/scripts/install-sg-exim.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@ function exim4_setup {
   sed -i "s/dc_other_hostnames=.*$/dc_other_hostnames='$DOMAIN'/" /etc/exim4/update-exim4.conf.conf
   sed -i "s/dc_local_interfaces=.*$/dc_local_interfaces='$IPADDRESS'/" /etc/exim4/update-exim4.conf.conf
   sed -i "s/dc_readhost=.*$/dc_readhost=''/" /etc/exim4/update-exim4.conf.conf
-  sed -i "s/dc_relay_domains=.*$/dc_relay_domains='ob."$DOMAIN"'/" /etc/exim4/update-exim4.conf.conf
+  sed -i "s/dc_relay_domains=.*$/dc_relay_domains='ob.$DOMAIN'/" /etc/exim4/update-exim4.conf.conf
```
Comment (fredericmora) [inline spamgourmet-clone/scripts/install-sg-exim.sh:30]: Regarding this method for creating configuration file, I also have a few configs modified on the fly with `sed`. I recently had a problem where a developer changed the config slightly in a way that made some of the `sed` commands fail. The problem can be avoided by using some kind of templating, e.g. Perl Text::Template. Then the config files with the templating expressions are rename to `.conf.in` and the templates are cleanly replaced to produce the `.conf` without risks.

Thoughts?

**Review by fredericmora — APPROVED** [review]

File: spamgourmet-clone/scripts/config-certs-provisioning.sh (hunk 1 above)
Comment (mikedlr) [reply spamgourmet-clone/scripts/config-certs-provisioning.sh:8]: Right now it's just a naughty, easily searchable magic number to make it different from other reasons for failure. The LDP defines a range from 64 - 113 as the good ones to use so we at least fit that. 

https://tldp.org/LDP/abs/html/exitcodes.html

we can get around to regularizing this some time soon.

File: spamgourmet-clone/scripts/install-sg-exim.sh (hunk 2 above)
Comment (mikedlr) [reply spamgourmet-clone/scripts/install-sg-exim.sh:30]: I've added this to the existing ticket to clean up templating. https://github.com/spamgourmet/spamgourmet/issues/52

**Pull request merged**